iot-hub/Tutorials/ConnectivityTests/SimulatedDevice-2.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// A test SAS token. To generate using the CLI:
+// az iot hub generate-sas-token --device-id MyTestDevice --hub-name {YourIoTHubName}
+var test_token = '{your test token}'
+
+// Using the Node.js Device SDK for IoT Hub:
+//   https://github.com/Azure/azure-iot-sdk-node
+// The sample connects to a device-specific MQTT endpoint on your IoT Hub.
+
+var Protocol = require('azure-iot-device-mqtt').Mqtt;
+var Client = require('azure-iot-device').Client;
+var Message = require('azure-iot-device').Message;
+var NoRetry = require('azure-iot-common').NoRetry;
+var crypto = require('crypto');
+
+// Example code showing how to generate a SAS token.
+var generateSasToken = function(resourceUri, signingKey, policyName, expiresInMins) {
+  resourceUri = encodeURIComponent(resourceUri);
+
+  // Set expiration in seconds
+  var expires = (Date.now() / 1000) + expiresInMins * 60;
+  expires = Math.ceil(expires);
+  var toSign = resourceUri + '\n' + expires;
+
+  // Use crypto
+  var hmac = crypto.createHmac('sha256', new Buffer(signingKey, 'base64'));
+  hmac.update(toSign);
+  var base64UriEncoded = encodeURIComponent(hmac.digest('base64'));
+
+  // Construct autorization string
+  var token = "SharedAccessSignature sr=" + resourceUri + "&sig="
+  + base64UriEncoded + "&se=" + expires;
+  if (policyName) token += "&skn="+policyName;
+  return token;
+};
```

Comment:
the SDK has APIs to do this (in the SharedAccessSignature and ConnectionString classes)